airflow/providers/microsoft/azure/hooks/asb.py

```diff
@@ -0,0 +1,203 @@
+
+    def get_conn(self):
+        raise NotImplementedError
+
+
+class AdminClientHook(BaseAzureServiceBusHook):
+    """
+    Interacts with ServiceBusAdministrationClient client
+    to create, update, list, and delete resources of a
+    Service Bus namespace.  This hook uses the same Azure Service Bus client connection inherited
+    from the base class
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    def get_conn(self) -> ServiceBusAdministrationClient:
+        """
+        Create and returns ServiceBusAdministrationClient by using the connection
+        string in connection details
+        """
+        conn = self.get_connection(self.conn_id)
+
+        connection_string: str = str(conn.schema)
+        return ServiceBusAdministrationClient.from_connection_string(connection_string)
+
+    def create_queue(
+        self,
+        queue_name: str,
+        max_delivery_count: int = 10,
+        dead_lettering_on_message_expiration: bool = True,
+        enable_batched_operations: bool = True,
+    ) -> QueueProperties:
+        """
+        Create Queue by connecting to service Bus Admin client return the QueueProperties
+
+        :param queue_name: The name of the queue or a QueueProperties with name.
+        :param max_delivery_count: The maximum delivery count. A message is automatically
+            dead lettered after this number of deliveries. Default value is 10..
+        :param dead_lettering_on_message_expiration: A value that indicates whether this subscription has
+            dead letter support when a message expires.
+        :param enable_batched_operations: Value that indicates whether server-side batched
+            operations are enabled.
+        """
+        if queue_name is None:
+            raise TypeError("Queue name cannot be None.")
+
+        with self.get_conn() as service_mgmt_conn:
+            queue = service_mgmt_conn.create_queue(
+                queue_name,
+                max_delivery_count=max_delivery_count,
+                dead_lettering_on_message_expiration=dead_lettering_on_message_expiration,
+                enable_batched_operations=enable_batched_operations,
+            )
+            return queue
+
+    def delete_queue(self, queue_name: str) -> None:
+        """
+        Delete the queue by queue_name in service bus namespace
+
+        :param queue_name: The name of the queue or a QueueProperties with name.
+        """
+        if queue_name is None:
+            raise TypeError("Queue name cannot be None.")
+
+        with self.get_conn() as service_mgmt_conn:
+            service_mgmt_conn.delete_queue(queue_name)
+
+
+class MessageHook(BaseAzureServiceBusHook):
+    """
+    Interacts with ServiceBusClient and acts as a high level interface
+    for getting ServiceBusSender and ServiceBusReceiver.
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
```

Comment:
Do we even need an `__init__` if we are just passing every arg, kwarg as they are to super class?

Reply:
Yes, we don't need it